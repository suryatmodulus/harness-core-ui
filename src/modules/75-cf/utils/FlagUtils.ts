/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import { Feature, GitDetails, PatchFeatureQueryParams, usePatchFeature } from 'services/cf'

export interface FlagPatchParams {
  accountIdentifier: string
  orgIdentifier: string
  projectIdentifier: string
  environmentIdentifier: string
}

const makeInstruction = (kind: string, variation: string, targetIdentifiers: string[], gitDetails?: GitDetails) => {
  const instructions = {
    instructions: [
      {
        kind,
        parameters: {
          variation,
          targets: targetIdentifiers
        }
      }
    ]
  }

  return gitDetails ? { ...instructions, gitDetails } : instructions
}

const makePatchHook =
  (patchKind: string) =>
  ({ accountIdentifier, orgIdentifier, projectIdentifier, environmentIdentifier }: FlagPatchParams) => {
    const { mutate } = usePatchFeature({
      identifier: '',
      queryParams: {
        account: accountIdentifier,
        accountIdentifier,
        org: orgIdentifier,
        project: projectIdentifier,
        environment: environmentIdentifier
      } as PatchFeatureQueryParams
    })

    return (featureFlag: Feature, variation: any, targetIdentifiers: string[], gitDetails?: GitDetails) => {
      const body = makeInstruction(patchKind, variation, targetIdentifiers, gitDetails)

      return mutate(body, { pathParams: { identifier: featureFlag.identifier } })
    }
  }

export const useServeFeatureFlagVariationToTargets = makePatchHook('addTargetsToVariationTargetMap')

export const useRemoveFeatureFlagVariationFromTargets = makePatchHook('removeTargetsToVariationTargetMap')
