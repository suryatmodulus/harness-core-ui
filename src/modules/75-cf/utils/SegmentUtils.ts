/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import { PatchSegmentQueryParams, usePatchSegment } from 'services/cf'

export interface TargetSegmentParams {
  accountIdentifier: string
  orgIdentifier: string
  environmentIdentifier: string
  projectIdentifier: string
}

const makeInstructions = (kind: string, targetIdentifiers: string[]) => ({
  instructions: [
    {
      kind,
      parameters: {
        targets: targetIdentifiers
      }
    }
  ]
})

const makeTargetsToSegmentHook =
  (patchKind: string) =>
  ({ accountIdentifier, orgIdentifier, environmentIdentifier, projectIdentifier }: TargetSegmentParams) => {
    const { mutate } = usePatchSegment({
      identifier: '',
      queryParams: {
        account: accountIdentifier,
        accountIdentifier,
        org: orgIdentifier,
        project: projectIdentifier,
        environment: environmentIdentifier
      } as PatchSegmentQueryParams
    })

    return (segmentIdentifier: string, targetIdentifiers: string[]) => {
      const body = makeInstructions(patchKind, targetIdentifiers)
      return mutate(body, { pathParams: { identifier: segmentIdentifier } })
    }
  }

/** Add targets to an include segment list */
export const useAddTargetsToIncludeList = makeTargetsToSegmentHook('addToIncludeList')

/** Remove targets to an include segment list */
export const useRemoveTargetsFromIncludeList = makeTargetsToSegmentHook('removeFromIncludeList')

/** Add targets to an exclude segment list */
export const useAddTargetsToExcludeList = makeTargetsToSegmentHook('addToExcludeList')

/** Remove targets to an exclude segment list */
export const useRemoveTargetsFromExcludeList = makeTargetsToSegmentHook('removeFromExcludeList')
