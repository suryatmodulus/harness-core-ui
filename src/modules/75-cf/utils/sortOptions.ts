/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

export interface Option {
  label: string
  value: string
}

export const sortOptions = (options: Option[]): Option[] =>
  options.sort(({ label: a }, { label: b }) => a.localeCompare(b))
