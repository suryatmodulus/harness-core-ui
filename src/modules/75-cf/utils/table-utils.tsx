/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import type { CellProps } from 'react-table'

export function withTableData<Data extends Record<any, any>, T>(
  mapDataToProps: (data: CellProps<Data>) => T
): (Comp: React.FC<T>) => React.FC<CellProps<Data>> {
  return (Comp: React.FC<T>) => (cellProps: CellProps<Data>) => <Comp {...mapDataToProps(cellProps)} />
}
