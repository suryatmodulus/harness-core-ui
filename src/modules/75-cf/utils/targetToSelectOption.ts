import type { SelectOption } from '@wings-software/uicore'
import type { Target } from 'services/cf'

export default function targetToSelectOption({ name, identifier }: Target): SelectOption {
  return { label: name, value: identifier }
}
