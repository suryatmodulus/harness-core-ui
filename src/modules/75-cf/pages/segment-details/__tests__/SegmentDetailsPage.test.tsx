/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { fireEvent, getAllByText, getByText, render, waitFor } from '@testing-library/react'
import { TestWrapper } from '@common/utils/testUtils'
import mockImport from 'framework/utils/mockImport'
import { SegmentDetailPage } from '../SegmentDetailPage'
import mockSegment from './mockSegment'
import mockTargets from './mockTargets'
import mockEnvironment from '../../environment-details/__tests__/mockEnvironment'

jest.mock('@cf/hooks/useGitSync', () => ({
  useGitSync: jest.fn(() => ({
    getGitSyncFormMeta: jest.fn(() => ({
      gitSyncInitialValues: {},
      gitSyncValidationSchema: {}
    })),
    isAutoCommitEnabled: false,
    isGitSyncEnabled: true,
    isGitSyncActionsEnabled: true,
    handleAutoCommit: jest.fn()
  }))
}))

describe('SegmentDetailsPage', () => {
  test('SegmentDetailsPage should render loading correctly', async () => {
    mockImport('services/cf', {
      useGetSegment: () => ({
        data: undefined,
        loading: true,
        error: undefined,
        refetch: jest.fn()
      })
    })

    mockImport('@cf/hooks/useSyncedEnvironment', {
      useSyncedEnvironment: () => ({ loading: true, refetch: jest.fn() })
    })

    const { container } = render(
      <TestWrapper
        path="/account/:accountId/cf/orgs/:orgIdentifier/projects/:projectIdentifier/onboarding/detail"
        pathParams={{ accountId: 'dummy', orgIdentifier: 'dummy', projectIdentifier: 'dummy' }}
      >
        <SegmentDetailPage />
      </TestWrapper>
    )

    expect(container.querySelector('.bp3-spinner')).toBeDefined()
  })

  test('SegmentDetailsPage should render error correctly 1', async () => {
    const error = { message: 'SOME ERROR OCCURS' }

    mockImport('services/cf', {
      useGetSegment: () => ({
        data: undefined,
        loading: undefined,
        error,
        refetch: jest.fn()
      })
    })

    mockImport('@cf/hooks/useSyncedEnvironment', {
      useSyncedEnvironment: () => ({ data: mockEnvironment, loading: undefined, error: undefined, refetch: jest.fn() })
    })

    const { container } = render(
      <TestWrapper
        path="/account/:accountId/cf/orgs/:orgIdentifier/projects/:projectIdentifier/onboarding/detail"
        pathParams={{ accountId: 'dummy', orgIdentifier: 'dummy', projectIdentifier: 'dummy' }}
      >
        <SegmentDetailPage />
      </TestWrapper>
    )

    expect(getAllByText(container, error.message)).toBeDefined()
  })

  test('SegmentDetailsPage should render error correctly 2', async () => {
    const error = { message: 'SOME ERROR OCCURS' }

    mockImport('services/cf', {
      useGetSegment: () => ({
        data: mockSegment,
        loading: undefined,
        error: undefined,
        refetch: jest.fn()
      })
    })

    mockImport('@cf/hooks/useSyncedEnvironment', {
      useSyncedEnvironment: () => ({ data: undefined, loading: undefined, error, refetch: jest.fn() })
    })

    const { container } = render(
      <TestWrapper
        path="/account/:accountId/cf/orgs/:orgIdentifier/projects/:projectIdentifier/onboarding/detail"
        pathParams={{ accountId: 'dummy', orgIdentifier: 'dummy', projectIdentifier: 'dummy' }}
      >
        <SegmentDetailPage />
      </TestWrapper>
    )

    expect(getAllByText(container, error.message)).toBeDefined()
  })

  test('SegmentDetailsPage should render data correctly', async () => {
    mockImport('services/cf', {
      useGetSegment: () => ({
        data: mockSegment,
        loading: undefined,
        error: undefined,
        refetch: jest.fn()
      }),

      useGetSegmentFlags: () => ({
        data: []
      }),

      useGetAllTargets: () => ({
        data: mockTargets
      })
    })

    mockImport('@cf/hooks/useSyncedEnvironment', {
      useSyncedEnvironment: () => ({ data: mockEnvironment, loading: undefined, error: undefined, refetch: jest.fn() })
    })

    const { container } = render(
      <TestWrapper
        path="/account/:accountId/cf/orgs/:orgIdentifier/projects/:projectIdentifier/onboarding/detail"
        pathParams={{ accountId: 'dummy', orgIdentifier: 'dummy', projectIdentifier: 'dummy' }}
      >
        <SegmentDetailPage />
      </TestWrapper>
    )

    expect(getAllByText(container, mockSegment.name)).toBeDefined()
    expect(container).toMatchSnapshot()
  })

  test('Delete function should work', async () => {
    const mutate = jest.fn(() => {
      return Promise.resolve({ data: {} })
    })

    mockImport('services/cf', {
      useGetSegment: () => ({
        data: mockSegment,
        loading: undefined,
        error: undefined,
        refetch: jest.fn()
      }),

      useGetSegmentFlags: () => ({
        data: []
      }),

      useGetAllTargets: () => ({
        data: mockTargets
      }),

      useDeleteSegment: () => ({
        mutate
      })
    })

    mockImport('@cf/hooks/useSyncedEnvironment', {
      useSyncedEnvironment: () => ({ data: mockEnvironment, loading: undefined, error: undefined, refetch: jest.fn() })
    })

    const { container } = render(
      <TestWrapper
        path="/account/:accountId/cf/orgs/:orgIdentifier/projects/:projectIdentifier/onboarding/detail"
        pathParams={{ accountId: 'dummy', orgIdentifier: 'dummy', projectIdentifier: 'dummy' }}
      >
        <SegmentDetailPage />
      </TestWrapper>
    )

    expect(getAllByText(container, mockSegment.name)).toBeDefined()
    ;(document.querySelector('[data-icon="Options"]') as HTMLButtonElement)?.click()
    ;(document.querySelector('[icon="cross"]') as HTMLButtonElement)?.click()

    expect(getByText(document.body, 'confirm')).toBeDefined()
    fireEvent.click(getByText(document.body, 'confirm') as HTMLButtonElement)
    await waitFor(() => expect(mutate).toBeCalledTimes(1))
  })
})
