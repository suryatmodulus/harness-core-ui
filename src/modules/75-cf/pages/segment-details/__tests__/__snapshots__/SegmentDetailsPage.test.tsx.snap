// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SegmentDetailsPage SegmentDetailsPage should render data correctly 1`] = `
<div>
  <div
    class="StyledProps--font StyledProps--main StyledProps--padding-top-large StyledProps--padding-right-xlarge StyledProps--padding-bottom-large StyledProps--padding-left-xlarge"
    style="background: rgba(219, 241, 255, 0.46); position: relative; height: 120px;"
  >
    <div
      class="Layout--vertical Layout--layout-spacing-medium StyledProps--main"
    >
      <div
        class="Layout--horizontal StyledProps--main Breadcrumbs--breadcrumbs StyledProps--flex StyledProps--flex-align-center-center StyledProps--flex-justifyContent-flex-start"
      >
        <div
          class="Layout--horizontal StyledProps--main StyledProps--flex StyledProps--flex-align-center-center StyledProps--flex-justifyContent-flex-start"
        >
          <a
            class="Breadcrumbs--breadcrumb"
            href="/account/dummy/cf/orgs/dummy/projects/dummy"
          >
            <span
              class="bp3-icon StyledProps--main StyledProps--padding-right-xsmall"
              data-icon="cf-main"
            >
              <svg
                height="16"
                viewBox="0 0 32 32"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M10.632 2.08A5.19 5.19 0 004.62 6.29L2.08 20.697a5.19 5.19 0 004.21 6.012l7.464 1.316a.2.2 0 00.207-.298L6.882 15.656c-.296-.469-.292-1.382.05-1.6.08-.05.313-.259.313-.259l.398-.436.066-.072c.37-.406.704-.773 1.76-1.302l.227-.079c1.548-.484 2.41-.449 3.27-.414.862.035 1.724.07 3.274-.414 1.272-.397 2.43-1.178 3.074-1.667a.622.622 0 01.9.15l4.596 7.362a.53.53 0 01-.101.696 8.694 8.694 0 01-2.834 1.538c-1.549.484-2.432.463-3.316.442-.884-.021-1.768-.042-3.318.442-1.314.41-1.868.948-2.07 1.204-.048.061-.07.21-.032.279.4.724 2.295 4.06 3.794 6.693a.998.998 0 00.695.49l3.07.541a5.19 5.19 0 006.012-4.21l2.54-14.408a5.19 5.19 0 00-4.21-6.012L10.633 2.08z"
                  fill="url(#cf-main_svg__a)"
                />
                <path
                  d="M16.646 13.074c-1.303.34-2.093.316-2.882.29-.79-.024-1.58-.049-2.881.291-.744.216-1.178.592-1.587 1.072a.197.197 0 00-.018.23l2.482 4.069c.064.106.209.13.307.055.374-.283 1.28-.9 2.42-1.198 1.469-.383 2.348-.375 3.273-.366.717.007 1.462.014 2.53-.16a4.454 4.454 0 001.629-.635.347.347 0 00.094-.488L19.472 12.3a.421.421 0 00-.56-.128 9.57 9.57 0 01-2.266.902z"
                  fill="url(#cf-main_svg__b)"
                />
                <defs>
                  <lineargradient
                    gradientUnits="userSpaceOnUse"
                    id="cf-main_svg__a"
                    x1="6.837"
                    x2="23.78"
                    y1="2"
                    y2="28.227"
                  >
                    <stop
                      stop-color="#FFC100"
                    />
                    <stop
                      offset="1"
                      stop-color="#E66B35"
                    />
                  </lineargradient>
                  <lineargradient
                    gradientUnits="userSpaceOnUse"
                    id="cf-main_svg__b"
                    x1="6.837"
                    x2="23.78"
                    y1="2"
                    y2="28.227"
                  >
                    <stop
                      stop-color="#FFC100"
                    />
                    <stop
                      offset="1"
                      stop-color="#E66B35"
                    />
                  </lineargradient>
                </defs>
              </svg>
            </span>
            <p
              class="StyledProps--font StyledProps--main StyledProps--intent-primary StyledProps--font-size-small"
            >
              dummy
            </p>
          </a>
          <span
            class="bp3-icon StyledProps--main StyledProps--color StyledProps--color-grey500 StyledProps--padding-right-xsmall StyledProps--padding-left-xsmall"
            data-icon="main-chevron-right"
          >
            <svg
              height="8"
              viewBox="0 0 8 14"
              width="8"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M7.7 6.3l-6-6C1.5.1 1.3 0 1 0 .4 0 0 .4 0 1c0 .3.1.5.3.7L5.6 7 .3 12.3c-.2.2-.3.4-.3.7 0 .6.4 1 1 1 .3 0 .5-.1.7-.3l6-6c.2-.2.3-.4.3-.7s-.1-.5-.3-.7z"
                fill="#4d5969"
              />
            </svg>
          </span>
        </div>
        <div
          class="Layout--horizontal StyledProps--main StyledProps--flex StyledProps--flex-align-center-center StyledProps--flex-justifyContent-flex-start"
        >
          <a
            class="Breadcrumbs--breadcrumb"
            href="/account/dummy/cf/orgs/dummy/projects/dummy/target-management/target-groups"
          >
            <p
              class="StyledProps--font StyledProps--main StyledProps--intent-primary StyledProps--font-size-small"
            >
              cf.shared.targetManagement: cf.shared.segments
            </p>
          </a>
          <span
            class="bp3-icon StyledProps--main StyledProps--color StyledProps--color-grey500 StyledProps--padding-right-xsmall StyledProps--padding-left-xsmall"
            data-icon="main-chevron-right"
          >
            <svg
              height="8"
              viewBox="0 0 8 14"
              width="8"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M7.7 6.3l-6-6C1.5.1 1.3 0 1 0 .4 0 0 .4 0 1c0 .3.1.5.3.7L5.6 7 .3 12.3c-.2.2-.3.4-.3.7 0 .6.4 1 1 1 .3 0 .5-.1.7-.3l6-6c.2-.2.3-.4.3-.7s-.1-.5-.3-.7z"
                fill="#4d5969"
              />
            </svg>
          </span>
        </div>
      </div>
      <div
        class="Layout--horizontal Layout--layout-spacing-small StyledProps--main"
        style="margin-left: -5px;"
      >
        <div
          class="StyledProps--font StyledProps--main"
          style="border-radius: 50%; overflow: hidden;"
        >
          <div
            class="avatar Avatar--Avatar Avatar--contentStyle"
            style="cursor: default;"
          >
            <span
              class="bp3-popover-wrapper Avatar--avatarPopOver"
            >
              <span
                class="bp3-popover-target"
              >
                <div
                  class="Avatar--AvatarInner"
                  style="border-radius: 100%; line-height: 42px; width: 48px; height: 48px; font-size: 18px;"
                  tabindex="0"
                >
                  S
                </div>
              </span>
            </span>
          </div>
        </div>
        <div
          class="StyledProps--font StyledProps--main"
        >
          <div
            class="Layout--horizontal Layout--layout-spacing-medium StyledProps--main"
          >
            <p
              class="StyledProps--font StyledProps--main"
              style="font-size: 20px;"
            >
              segment1
            </p>
            <span
              class="StyledProps--font StyledProps--main StyledProps--inline"
              style="background-color: rgb(239, 251, 255); margin-bottom: 0px; padding: 4px 8px; border-radius: 4px; font-size: 12px; line-height: 15px; color: rgb(34, 34, 42); border: 1px solid #d9dae6;"
            >
              idLabel
              segment1
              <button
                style="background: none; padding: 0px; transform: translateY(-2px); cursor: pointer;"
                title="clickToCopy"
              >
                <span
                  class="bp3-icon bp3-icon-duplicate StyledProps--main StyledProps--color StyledProps--color-grey350"
                  icon="duplicate"
                >
                  <svg
                    data-icon="duplicate"
                    height="12"
                    viewBox="0 0 16 16"
                    width="12"
                  >
                    <desc>
                      duplicate
                    </desc>
                    <path
                      d="M15 0H5c-.55 0-1 .45-1 1v2h2V2h8v7h-1v2h2c.55 0 1-.45 1-1V1c0-.55-.45-1-1-1zm-4 4H1c-.55 0-1 .45-1 1v10c0 .55.45 1 1 1h10c.55 0 1-.45 1-1V5c0-.55-.45-1-1-1zm-1 10H2V6h8v8z"
                      fill-rule="evenodd"
                    />
                  </svg>
                </span>
              </button>
            </span>
          </div>
          <p
            class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-grey400 StyledProps--padding-top-xsmall"
          >
            cf.targetDetail.createdOnDate
          </p>
        </div>
      </div>
    </div>
    <div
      class="StyledProps--font StyledProps--main"
      style="position: absolute; top: 15px; right: 25px;"
    >
      <span
        class="bp3-popover-wrapper"
      >
        <span
          class="bp3-popover-target"
        >
          <button
            class="bp3-button bp3-minimal Button--button StyledProps--font StyledProps--main Button--with-current-color Button--iconOnly Button--without-shadow Button--withLeftIcon"
            type="button"
          >
            <span
              class="bp3-icon StyledProps--main"
              data-icon="Options"
            >
              <svg
                fill="none"
                height="16"
                viewBox="0 0 4 16"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M2 12.5a1 1 0 100 2 1 1 0 000-2zm-2 1a2 2 0 114 0 2 2 0 01-4 0zm2-12a1 1 0 100 2 1 1 0 000-2zm-2 1a2 2 0 114 0 2 2 0 01-4 0zM2 7a1 1 0 100 2 1 1 0 000-2zM0 8a2 2 0 114 0 2 2 0 01-4 0z"
                  fill="currentColor"
                  fill-rule="evenodd"
                />
              </svg>
            </span>
          </button>
        </span>
      </span>
    </div>
    <p
      class="StyledProps--font StyledProps--main"
      style="position: absolute; top: 76px; right: 30px;"
    >
      <span>
        cf.targetDetail.environmentLine
      </span>
    </p>
  </div>
  <div
    class="StyledProps--font StyledProps--main"
    style="height: calc(100% - 120px); overflow: auto; background: rgba(228, 235, 244, 0.2);"
  >
    <div
      class="Layout--vertical StyledProps--main"
      style="flex-grow: 1; height: 100%;"
    >
      <div
        class="StyledProps--font StyledProps--main StyledProps--border StyledProps--border-bottom StyledProps--border-color-grey200"
      >
        <div
          class="StyledProps--font StyledProps--main StyledProps--flex StyledProps--flex-justifyContent-space-between StyledProps--flex-alignItems-center StyledProps--margin-left-xxlarge StyledProps--margin-top-large StyledProps--margin-bottom-large"
          style="width: 230px;"
        >
          <div
            class="Layout--horizontal Layout--layout-spacing-small StyledProps--main"
            style="width: 400px;"
          >
            <div
              class="StyledProps--font StyledProps--main gitRepoText"
            >
              <span
                class="bp3-icon StyledProps--main"
                data-icon="repository"
              >
                <svg
                  height="16"
                  viewBox="0 0 13 14"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M12.438 2.452l-.006-.017a.367.367 0 00-.067-.114l-.004-.007L10.409.125a.374.374 0 00-.28-.125H2.327a.376.376 0 00-.28.125L.095 2.314l-.005.008a.374.374 0 00-.066.113l-.006.017a.37.37 0 00-.018.11v11.064c0 .206.168.374.374.374h11.707a.375.375 0 00.375-.374V2.563a.369.369 0 00-.018-.111zm-7.91 7.86h3.4v.357a.318.318 0 01-.318.318H4.846a.318.318 0 01-.318-.318v-.357zM.75 9.563V6.625h3.03v.357c0 .588.479 1.066 1.067 1.066H7.61c.588 0 1.066-.478 1.066-1.066v-.357h3.03v2.938H.75zm3.03-6.626v.357c0 .588.479 1.067 1.067 1.067H7.61c.589 0 1.067-.479 1.067-1.067v-.357h3.03v2.939H.75V2.937h3.03zm4.15 0v.357a.318.318 0 01-.318.318H4.846a.318.318 0 01-.318-.318v-.357h3.4zm-3.4 3.688h3.4v.357a.318.318 0 01-.319.317H4.846a.317.317 0 01-.318-.317v-.357zM2.493.749h7.468l1.284 1.44H1.21L2.494.748zm9.213 12.502H.75v-2.939h3.03v.357c0 .589.479 1.067 1.067 1.067H7.61c.588 0 1.066-.478 1.066-1.067v-.357h3.03v2.94z"
                    fill="currentColor"
                  />
                </svg>
              </span>
              <p
                class="StyledProps--font Text--lineclamp StyledProps--main StyledProps--color StyledProps--color-black"
                style="--text-line-clamp: 1;"
              >
                
              </p>
            </div>
            <div
              class="StyledProps--font StyledProps--main"
            >
              <div
                class="bp3-popover-wrapper bp3-fill"
              >
                <div
                  class="bp3-popover-target"
                >
                  <button
                    class="branchActionButton"
                  >
                    <div
                      class="StyledProps--font StyledProps--main branchActionButtonWrapper"
                    >
                      <span
                        class="bp3-icon StyledProps--main"
                        data-icon="git-new-branch"
                      >
                        <svg
                          height="15"
                          viewBox="0 0 13 16"
                          width="15"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M10.406.003a2.399 2.399 0 00-.8 4.658v.94a.8.8 0 01-.8.8H4.001a2.413 2.413 0 00-.8.14V4.66A2.399 2.399 0 002.402 0a2.399 2.399 0 00-.8 4.661v6.678a2.399 2.399 0 00.8 4.661 2.399 2.399 0 00.8-4.661v-2.54a.8.8 0 01.8-.799h4.803a2.4 2.4 0 002.402-2.4v-.939a2.399 2.399 0 00-.8-4.658z"
                            fill="currentColor"
                          />
                        </svg>
                      </span>
                      <p
                        class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-grey900"
                      >
                        
                      </p>
                      <div
                        class="StyledProps--font StyledProps--main gitStatusIcon"
                      >
                        <span
                          class="bp3-popover-wrapper"
                        >
                          <span
                            class="bp3-popover-target"
                          >
                            <span
                              class="bp3-icon bp3-icon-full-circle StyledProps--main autoCommitDisabled"
                              data-testid="auto-commit-status-icon"
                              icon="full-circle"
                              tabindex="0"
                            >
                              <svg
                                data-icon="full-circle"
                                height="10"
                                viewBox="0 0 16 16"
                                width="10"
                              >
                                <desc>
                                  full-circle
                                </desc>
                                <path
                                  d="M8 0a8 8 0 100 16A8 8 0 108 0z"
                                  fill-rule="evenodd"
                                />
                              </svg>
                            </span>
                          </span>
                        </span>
                      </div>
                    </div>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="Layout--horizontal StyledProps--main"
        style="height: 100%;"
      >
        <div
          class="StyledProps--font StyledProps--main"
          style="background: rgb(248, 250, 251); overflow: auto; min-width: 480px; width: 480px; height: 100%;"
        >
          <div
            class="Layout--horizontal StyledProps--main StyledProps--padding-right-small"
            style="align-items: baseline; width: 100%; height: 55px;"
          >
            <h3
              class="StyledProps--font StyledProps--font-h3 StyledProps--main StyledProps--padding-xxlarge"
              level="3"
              style="font-size: 14px; color: rgb(34, 39, 45); font-weight: 600; padding-bottom: 0px;"
            >
              cf.segments.usingSegment
            </h3>
            <span
              style="flex-grow: 1;"
            />
            <button
              class="bp3-button bp3-minimal Button--button StyledProps--font StyledProps--main StyledProps--intent-primary Button--with-current-color Button--without-shadow"
              data-testid="add-feature-flags-button"
              type="button"
            >
              <span
                class="bp3-button-text"
              >
                cf.segmentDetail.addToFlag
              </span>
            </button>
          </div>
          <div
            class="StyledProps--font StyledProps--main"
            style="overflow: auto; width: 100%; height: calc(100% - 55px);"
          >
            <div
              class="StyledProps--font StyledProps--main StyledProps--padding-medium StyledProps--margin-top-xxlarge"
              style="box-shadow: 0px 0px 1px rgba(40, 41, 61, 0.08), 0px 0.5px 2px rgba(96, 97, 112, 0.16); border-radius: 8px;"
            >
              <p
                class="StyledProps--font StyledProps--main StyledProps--background-white StyledProps--padding-xsmall StyledProps--color StyledProps--color-grey400"
                style="text-align: center;"
              >
                cf.segmentDetail.noFlagsUseThisSegment
              </p>
            </div>
          </div>
        </div>
        <div
          class="StyledProps--font StyledProps--main"
          style="overflow: auto; background: rgb(252, 253, 253); width: 100%; height: 100%;"
        >
          <div
            class="StyledProps--font StyledProps--main tabContainer"
          >
            <div
              class="bp3-tabs Tabs--main Tabs--horizontal"
            >
              <div
                class="bp3-tab-list"
                role="tablist"
              >
                <div
                  class="bp3-tab-indicator-wrapper"
                  style="height: 0px; transform: translateX(0px) translateY(0px); width: 0px;"
                >
                  <div
                    class="bp3-tab-indicator"
                  />
                </div>
                <div
                  aria-controls="bp3-tab-panel_targetSettings_attributes"
                  aria-disabled="false"
                  aria-expanded="true"
                  aria-selected="true"
                  class="bp3-tab"
                  data-tab-id="attributes"
                  id="bp3-tab-title_targetSettings_attributes"
                  role="tab"
                  tabindex="0"
                >
                  <p
                    class="StyledProps--font StyledProps--main tabTitle"
                  >
                    <span
                      class="align"
                      data-tooltip-id="ff_segmentRules_heading"
                    >
                      cf.shared.rules
                    </span>
                  </p>
                </div>
                <div
                  aria-controls="bp3-tab-panel_targetSettings_segments"
                  aria-disabled="false"
                  aria-expanded="false"
                  aria-selected="false"
                  class="bp3-tab"
                  data-tab-id="segments"
                  id="bp3-tab-title_targetSettings_segments"
                  role="tab"
                  tabindex="0"
                >
                  <p
                    class="StyledProps--font StyledProps--main tabTitle"
                  >
                    activityLog
                  </p>
                </div>
              </div>
              <div
                aria-hidden="false"
                aria-labelledby="bp3-tab-title_targetSettings_attributes"
                class="bp3-tab-panel"
                id="bp3-tab-panel_targetSettings_attributes"
                role="tabpanel"
              >
                <div
                  class="Layout--vertical Layout--layout-spacing-medium StyledProps--main StyledProps--padding-large"
                >
                  <div
                    class="Layout--horizontal StyledProps--main"
                    style="align-items: center; justify-content: flex-end;"
                  >
                    <button
                      class="bp3-button Button--button StyledProps--font StyledProps--main Button--with-current-color Button--withLeftIcon"
                      type="button"
                    >
                      <span
                        class="bp3-icon bp3-icon-edit StyledProps--main StyledProps--padding-right-xsmall"
                        icon="edit"
                      >
                        <svg
                          data-icon="edit"
                          height="16"
                          viewBox="0 0 16 16"
                          width="16"
                        >
                          <desc>
                            edit
                          </desc>
                          <path
                            d="M3.25 10.26l2.47 2.47 6.69-6.69-2.46-2.48-6.7 6.7zM.99 14.99l3.86-1.39-2.46-2.44-1.4 3.83zm12.25-14c-.48 0-.92.2-1.24.51l-1.44 1.44 2.47 2.47 1.44-1.44c.32-.32.51-.75.51-1.24.01-.95-.77-1.74-1.74-1.74z"
                            fill-rule="evenodd"
                          />
                        </svg>
                      </span>
                      <span
                        class="bp3-button-text"
                      >
                        cf.featureFlags.rules.editRules
                      </span>
                    </button>
                  </div>
                  <div
                    class="bp3-card bp3-elevation-0 Card--card"
                    style="width: 100%;"
                  >
                    <div
                      class="Layout--vertical Layout--layout-spacing-small StyledProps--main"
                    >
                      <p
                        class="StyledProps--font StyledProps--main StyledProps--font-size-medium StyledProps--font-weight-bold"
                      >
                        Individual Targets
                      </p>
                      <div
                        class="Layout--horizontal StyledProps--main StyledProps--padding-small"
                        style="display: flex; align-items: center; justify-content: flex-start;"
                      >
                        <p
                          class="StyledProps--font StyledProps--main"
                        >
                          Include the following:
                        </p>
                         
                      </div>
                      <div
                        class="bp3-divider"
                      />
                      <div
                        class="Layout--horizontal StyledProps--main StyledProps--padding-small"
                        style="display: flex; align-items: center; justify-content: flex-start;"
                      >
                        <p
                          class="StyledProps--font StyledProps--main"
                        >
                          Exclude the following:
                        </p>
                         
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
